django/http/request.py

```diff
@@ -705,23 +710,45 @@ def __str__(self):
     def __repr__(self):
         return "<%s: %s>" % (self.__class__.__qualname__, self)
 
-    @property
-    def is_all_types(self):
-        return self.main_type == "*" and self.sub_type == "*"
+    @cached_property
+    def params(self):
+        params = self._params.copy()
+        params.pop("q", None)
```

Comment:
This was a breaking change for user code that was using `media_type.params.get("q")` (e.g., [Zulip](https://github.com/zulip/zulip/blob/b6f111cc0c9c5958a0f8b3d35245f7b6581c027d/zerver/views/upload.py#L204)). That code can now be rewritten using `media_type.quality`, but only in Django ≥ 5.2. Can we restore `q` to the `params` dictionary?

Reply:
It’s frustrating to have my own fixes rejected for changing non-public APIs (#18393), when they get unnecessarily broken like this even in stable patch releases. Couldn’t we have done this without changing `params`?

Reply:
I think this will not be too hard to do and something we can support.

Looking at the RFC:

> Each media-range might be followed by optional applicable media type parameters (e.g., [charset](https://httpwg.org/specs/rfc9110.html#charset)), followed by an optional "q" parameter for indicating a relative weight ([Section 12.4.2](https://httpwg.org/specs/rfc9110.html#quality.values)).

Perhaps we should have `MediaType.params` which include `q` and `MediaType.type_params` which doesn't include `q`

@andersk can you create a ticket so we don't lose any discussion in a PR?

Reply:
> Perhaps we should have MediaType.params which include q and MediaType.type_params which doesn't include q

That idea sounds great to me! Happy to get an implementation together once there's a ticket.

Reply:
I accepted the ticket and set it to be a release blocker. I have created https://github.com/django/django/pull/19548 to solve this.

I chose a slightly different name for the new property: `range_params`. I think this name matches better the spec, and also "MediaType.type_params" felt a bit redundant.